cli/src/main/java/com/devonfw/tools/ide/tool/LocalToolCommandlet.java

```diff
@@ -176,6 +176,73 @@ protected void postExtract(Path extractedDir) {
 
   }
 
+  @Override
+  public VersionIdentifier getInstalledVersion() {
+
+    return getInstalledVersion(this.context.getSoftwarePath().resolve(getName()));
+  }
+
+  /**
+   * @param toolPath the installation {@link Path} where to find the version file.
+   * @return the currently installed {@link VersionIdentifier version} of this tool or {@code null} if not installed.
+   */
+  protected VersionIdentifier getInstalledVersion(Path toolPath) {
+
+    if (!Files.isDirectory(toolPath)) {
+      this.context.debug("Tool {} not installed in {}", getName(), toolPath);
+      return null;
+    }
+    Path toolVersionFile = toolPath.resolve(IdeContext.FILE_SOFTWARE_VERSION);
+    if (!Files.exists(toolVersionFile)) {
+      Path legacyToolVersionFile = toolPath.resolve(IdeContext.FILE_LEGACY_SOFTWARE_VERSION);
+      if (Files.exists(legacyToolVersionFile)) {
+        toolVersionFile = legacyToolVersionFile;
+      } else {
+        this.context.warning("Tool {} is missing version file in {}", getName(), toolVersionFile);
+        return null;
+      }
+    }
+    try {
+      String version = Files.readString(toolVersionFile).trim();
+      return VersionIdentifier.of(version);
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to read file " + toolVersionFile, e);
+    }
+  }
+
+  @Override
+  public String getInstalledEdition() {
+
+    return getInstalledEdition(this.context.getSoftwarePath().resolve(getName()));
+  }
+
+  /**
+   * @param toolPath the installation {@link Path} where to find currently installed tool. The name of the parent
+   *     directory of the real path corresponding to the passed {@link Path path} must be the name of the edition.
+   * @return the installed edition of this tool or {@code null} if not installed.
+   */
+  public String getInstalledEdition(Path toolPath) {
+
+    if (!Files.isDirectory(toolPath)) {
+      this.context.debug("Tool {} not installed in {}", getName(), toolPath);
+      return null;
+    }
+    try {
+      String edition = toolPath.toRealPath().getParent().getFileName().toString();
```

Comment:
First of all if the software was installed by `devonfw-ide` it will not be a symlink or junction and therefore this will cause an odd result.
Second, I am not sure if `toRealPath()` will resolve the link both for symlinks and junctions. The JavaDoc is unclear regarding to OS specific things like junctions.
However, I get the expectation that `toRealPath()` will then resolve to something like `D:\projects\_ide\software\default\java\java\17.0.6_10` and the filename of the parent is `java` what is indeed the installed edition.
However if it was not installed via symlink/junction, the result would IMHO be `software`.
It is unlikely that a tool will ever have an edition named `software` but this looks a little fragile to me.
Can we also get the version and edition of a custom tool?
IMHO this is not yet implemented but there it may again be different...